core/src/main/resources/xsl/authenticated-by.xsl

```diff
@@ -0,0 +1,118 @@
+
+        <xsl:variable name="allAuthenticationMethods" as="xsd:string*">
+            <xsl:sequence select="$authenticationMethods"/>
+            <xsl:if test="@rax:authenticatedBy">
+                <xsl:sequence select="tokenize(@rax:authenticatedBy,' ')"/>
+            </xsl:if>
+        </xsl:variable>
+        <xsl:copy>
+            <!--
+                If this method is a copy generated by normalization and
+                it contains authenticationMethods, then you *do not* want
+                to continue to reference the source method.  This is
+                because the method is now going te be different than the
+                source!
+            -->
+            <xsl:choose>
+                <xsl:when test="count($allAuthenticationMethods) != 0">
+                    <xsl:apply-templates select="@*[not(local-name() = 'id' and namespace-uri() = 'http://docs.rackspace.com/api')]"/>
+                    <xsl:if test="not(wadl:request)">
+                        <wadl:request>
+                            <xsl:call-template name="generateAuthenticationMethods">
+                                <xsl:with-param name="authenticationMethods" select="$allAuthenticationMethods"/>
+                            </xsl:call-template>
+                        </wadl:request>
+                    </xsl:if>
+                </xsl:when>
+                <xsl:otherwise>
+                    <xsl:apply-templates select="@*"/>
+                </xsl:otherwise>
+            </xsl:choose>
+            <xsl:apply-templates select="node()">
+                <xsl:with-param name="authenticationMethods" select="$allAuthenticationMethods"/>
+            </xsl:apply-templates>
+        </xsl:copy>
+    </xsl:template>
+
+    <xsl:template match="wadl:request">
+        <xsl:param name="authenticationMethods" as="xsd:string*" select="()"/>
+        <xsl:copy>
+            <xsl:apply-templates select="@*"/>
+            <xsl:if test="count($authenticationMethods) != 0">
+                <xsl:call-template name="generateAuthenticationMethods">
+                    <xsl:with-param name="authenticationMethods" select="$authenticationMethods"/>
+                </xsl:call-template>
+            </xsl:if>
+            <xsl:apply-templates select="node()">
+                <xsl:with-param name="authenticationMethods" select="$authenticationMethods"/>
+            </xsl:apply-templates>
+        </xsl:copy>
+    </xsl:template>
+
+    <xsl:template name="generateAuthenticationMethods">
+        <xsl:param name="authenticationMethods" as="xsd:string*" select="()"/>
+        <xsl:if test="not('#all' = $authenticationMethods)">
+            <xsl:for-each select="$authenticationMethods">
+                <wadl:param name="X-AUTHENTICATED-BY" style="header" rax:code="401" rax:anyMatch="true"
+                            rax:message="Authentication method not allowed for this operation" type="xsd:string"
```

Comment:
Please add a more specific message here, something that specifies than another authentication mechanism is required.